core/trino-main/src/test/java/io/trino/operator/scalar/TestArrayHistogram.java

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import io.trino.sql.query.QueryAssertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+public class TestArrayHistogram
+{
+    private QueryAssertions assertions;
+
+    @BeforeAll
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterAll
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testBasic()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42]"))
+                .isEqualTo(Map.of(42, 1L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[42, 7]"))
+                .isEqualTo(Map.of(42, 1L, 7, 1L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42, 42, 42]"))
+                .isEqualTo(Map.of(42, 3L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY['a', 'b', 'a']"))
+                .isEqualTo(Map.of("a", 2L, "b", 1L));
+    }
+
+    @Test
+    public void testDuplicateKeys()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42, 42, 42]"))
+                .isEqualTo(Map.of(42, 3L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY['a', 'b', 'a']"))
+                .isEqualTo(Map.of("a", 2L, "b", 1L));
+    }
+
+    @Test
+    public void testEmpty()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[]"))
+                .isEqualTo(Map.of());
+    }
+
+    @Test
+    public void testNullsIgnored()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[null]"))
+                .isEqualTo(Map.of());
+
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[42, null]"))
+                .isEqualTo(Map.of(42, 1L));
+    }
+
+    @Test
+    public void testLargeArray()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42, 42, 42, 7, 1, 7, 1, 42, 7, 1, 1, 42, 7, 42, 1, 7, 2, 3, 7, 42, 42]"))
+                .isEqualTo(Map.of(1, 5L, 2, 1L, 3, 1L, 7, 6L, 42, 8L));
+    }
+
+    @Test
+    public void testTypes()
```

Comment:
This test is probably overkill, I've tested many of the different types available in Trino, but we could probably strip this down to only a few types. Let me know if you want it kept as is or if you want types cut from the test. The test takes just over 1 second to run on my box.

Reply:
IMO the more the merrier :). This is a good representation of what's possible and a good test to glean examples from.